### ISSUE TITLE
new rule: no-unnamed-default-export

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#1927]
Good:

```js
const foo = () => {};

export default foo;

```

Bad:

```js
export default () => {};

```

**Comment by LoiKos** [comment]
I see this rules as been added to CRA. I'm looking to understand why do you add this rules as for me there is strictly no benefit to do that as you can import defaut with whatever you want value. 

```js
// file.js
const foo = () => {};

export default foo;

// otherfile.js
import bar from './file.js'

```

**Comment by gajus** [comment]
Yeah, this would be for function only.

**Comment by gajus** [comment]
It is not only for readability though.

It is needed for certain functionality too, e.g.

https://github.com/vercel/next.js/issues/12891

**Comment by ljharb** [comment]
Components shouldn't be arrow functions anyways.

I'd be on board with a rule that required exported functions to be named, but i'd want the default be to disallow arrows entirely, and for an option to be required to allow for the "good" form in the OP.

**Comment by LoiKos** [comment]
I don't see your point @ljharb as this rules doesn't apply Component but all `js` files are concerned AFAIK.

**Comment by ljharb** [comment]
@LoiKos hm, yeah, fair point.

In the general case, then, relying on *inferred* names is what I'd consider a bad practice, and arrow functions aren't capable of having any other kind of name. So I'd still prefer the default behavior to prefer explicitly-named constructs, with an option to allow implicitly-named constructs.
